internal/template/funcmap.go

```diff
@@ -170,3 +171,9 @@ func isDotNotation(s string) (bool, error) {
 	// - data["foo"][123].bar["baz"].foo
 	return regexp.MatchString(`^([\w]+(\[(("\w+")|\d+)\])*\.[\w]+(\[(("\w+")|\d+)\])*)+$`, s)
 }
+
+// This function returns either "terraformPattern" if present, or the
```

Comment:
can you add some more documents around why is it needed? i.e. the validation pattern for tfengien config might be different from the validation pattern for terraform configs. Something like that.

Reply:
Done!